packets.go

```diff
@@ -761,40 +761,34 @@ func (rows *textRows) readRow(dest []driver.Value) error {
 	}
 
 	// RowSet Packet
-	var n int
-	var isNull bool
 	pos := 0
 
 	for i := range dest {
-		// Read bytes and convert to string
-		dest[i], isNull, n, err = readLengthEncodedString(data[pos:])
+		b, isNull, n, err := readLengthEncodedString(data[pos:])
+		if err != nil {
+			return err
+		}
 		pos += n
-		if err == nil {
-			if !isNull {
-				if !mc.parseTime {
+		switch {
+		case isNull:
+			dest[i] = nil
```

Comment:
Thanks for PR, but I prefer "early return" pattern.

```go
if isNull {
    dest[i] = nil
    continue
}

if mc.parseTime {
    switch rows.rs.columns[i].fieldType {
        // snip
    }
}
dest[i] = b
```

Reply:
Just updated to use more "early return" style, but still keep indent level low. See 317d8045364d7a1a2a3e788d0093d8e81ba18791.